server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -1067,11 +1068,22 @@ public Metadata.Builder writeTo(Metadata metadata,
                 indexNumberOfShards = indexMetadata.getNumberOfShards();
             }
 
+            Settings settings = Settings.builder()
+                .put(indexMetadata.getSettings())
+                // Override only the settings that might have changed
+                .put(tableParameters.filter(s -> !indexMetadata.getSettings().hasValue(s)))
```

Comment:
without the filtering we get issues here:
https://github.com/crate/crate/blob/master/server/src/main/java/org/elasticsearch/cluster/routing/allocation/IndexMetadataUpdater.java#L204

Reply:
After investigation and with the help of @mfussenegger (thank you!) this is the correct way to go.
indexMetadata can have different settings, for example `number_of_replicas` than the `tableParameters`, because they are calculated by `AllocationService#adaptAutoExpandReplicas()` on a per index level (one index for simple tables, multiple indices for partitioned ones), and the value in `DocTableInfo#tableParameters` shouldn't change.